src/train_gui.cpp

```diff
@@ -332,7 +339,7 @@ int GetTrainDetailsWndVScroll(VehicleID veh_id, TrainDetailsWindowTabs det_tab)
 	} else {
 		for (const Train *v = Train::Get(veh_id); v != NULL; v = v->GetNextVehicle()) {
 			GetCargoSummaryOfArticulatedVehicle(v, &_cargo_summary);
-			num += max(1u, _cargo_summary.Length());
+			num += max(1u, (unsigned)_cargo_summary.size());
```

Comment:
shouldn't this already be unsigned (or size_t) ? why the cast? + 1 below

Reply:
Yes, it will be unsigned already.

Reply:
Loss of precision is the reason